### ISSUE TITLE
Support binary message

### ISSUE THREAD

**Issue by mingfai** [cloudfoundry-community/java-nats#28]
NATS itself does support binary message 
https://github.com/derekcollison/nats/blob/master/spec/binary_msg_spec.rb

**Comment by thomasjaeckle** [comment]
Hi there.
I am currently running in a situation where I want to transmit a byte[] and wrapping it into a String somehow screws the message up.

What about the commit of @mingfai ? Could it after all get into the codebase?

Regards Thomas

**Comment by mingfai** [comment]
if your app is not performance sensitive, and you don't want to wait or patch the lib yourself, consider to encode your byte[] to base64 string

**Comment by mheath** [comment]
I agree that being able to send/receive binary messages would be a very
useful feature to have. I think it would take some effort to get right both
from an API perspective as well as from a Netty ByteBuf efficiency
perspective.
Since Cloud Foundry doesn't use binary messages, it's hard for me to
justify investing the time to support this feature. Additionally, Cloud
Foundry is moving from NATS to more capable distributed state management
tools such as Consul. In spite of all the effort I've put into making this
client both fast and stable, I see this as a good thing. I have learned the
hard way that pushing the management of distributed state to every client
using a loose messaging system like NATS is really hard to get right.

NATS, of course, still has valid use cases. However, by the end of the
year, I don't expect to be maintaining this client any longer. If anyone is
interested in taking over this project and moving it out of the Cloud
Foundry community org, please let me know.

I would be really interested to see somebody build Java based NATS
server... mostly to definitively demonstrate that while gnatsd is faster
than Ruby natsd, gnatsd is still slow compared to Netty. :)

-Mike
